specutils/fitting/fitmodels.py

```diff
@@ -0,0 +1,507 @@
+from __future__ import division
+
+import functools
+import operator
+
+import numpy as np
+import astropy.units as u
+
+from ..spectra.spectrum1d import Spectrum1D
+from ..manipulation.utils import excise_regions
+from astropy.modeling import fitting, Model, models
+
+
+__all__ = ['fit_lines']
+
+
+def fit_lines(spectrum, model, fitter=fitting.SimplexLSQFitter(),
+              exclude_regions=None, weights=None, window=None):
+    """
+    Fit the input models (initial conditions) to the spectrum.  Output will be
+    the same models with the parameters set based on the fitting.
+
+    spectrum, models -> models
+
+    Parameters
+    ----------
+    spectrum : Spectrum1D
+        The spectrum object overwhich the equivalent width will be calculated.
+
+    model: `~astropy.modeling.Model` or list of `~astropy.modeling.Model`
+        The model or list of models that contain the initial guess.
+
+    exclude_regions : list of `~specutils.SpectralRegion`
+        List of regions to exclude in the fitting.
+
+    weights : list  (NOT IMPLEMENTED YET)
+        List of weights to define importance of fitting regions.
+
+    Returns
+    -------
+    models : Compound model of `~astropy.modeling.Model`
+        A compound model of models with fitted parameters.
+
```

Comment:
Food for thought: a second return value that I think might be useful:  the *individual* models.  That is, it could be:

```
input_model = continuum + line1 + line2 + line3
fitted_compound_model, (fitcontinuum, fitline1, fitline2, fitline3) = fit_lines(..., input_model...)
```

this is a convenience, of course, as you can just extract the second argument from the compound model anyway.  But is it a one-liner to do that, or not?  If it is maybe we can just put it in the docs as an example... But I'm certain *most* users want the separate components in the end, even if for the actually fitting they want the combined model.

Reply:
I think, if anywhere, maybe in the docs. Someone who is using it at this level would probably already know they can just do a `fitcontinuum, fitline1, fitline2, fitline3 = fitted_compound_model`, no?

Reply:
Actually... *I* didn't know that.  That's great!  Then yes, I'm happy with that, but *definitely* needs to be in the docs.